bindgen/lib.rs

```diff
@@ -2733,16 +2770,22 @@ impl Bindings {
         Ok(Cow::Owned("rustfmt".into()))
     }
 
-    /// Checks if rustfmt_bindings is set and runs rustfmt on the string
-    fn rustfmt_generated_string<'a>(
+    /// Formats a token stream with the formatter set up in `BindgenOptions`.
+    fn format_tokens<'a>(
         &self,
-        source: &'a str,
-    ) -> io::Result<Cow<'a, str>> {
+        tokens: &proc_macro2::TokenStream,
+    ) -> io::Result<String> {
         let _t = time::Timer::new("rustfmt_generated_string")
             .with_output(self.options.time_phases);
 
-        if !self.options.rustfmt_bindings {
-            return Ok(Cow::Borrowed(source));
+        match self.options.formatter {
+            Formatter::None => return Ok(tokens.to_string()),
+
+            #[cfg(feature = "prettyplease")]
+            Formatter::Prettyplease => {
+                prettyplease::unparse(&syn::parse_quote!(#tokens));
```

Comment:
I'm not super familiar with this section of code but I feel like you want to return the `prettyplease`d string here rather than continue with the continuing down to [`Command`](https://doc.rust-lang.org/std/process/struct.Command.html) of rustfmt.

```suggestion
                return Ok(prettyplease::unparse(&syn::parse_quote!(#tokens)));
```